chord_metadata_service/chord/views_search.py

```diff
@@ -318,29 +335,42 @@ def phenopacket_query_results(query, params, options=None):
         if "add_field" in options:
             fields.append(options["add_field"])
 
-        # Results displayed as 4/5 columns:
-        # "individuals ID", "table ID" (optional), [Alternate ids list], number of experiments, [Biosamples list...]
-        return queryset.values(
-                *fields,
-                alternate_ids=Coalesce(F("subject__alternate_ids"), [])
-            ).annotate(
-                # Weird bug with Django 4.1 here: num_experiments must come before the use of ArrayAgg or biosamples
-                # is considered as an ArrayField...
-                num_experiments=Count("biosamples__experiment"),
-                # Postgre specific: aggregates multiple values in a list
-                biosamples=Coalesce(
-                    ArrayAgg("biosamples__id", distinct=True, filter=Q(biosamples__id__isnull=False)),
-                    []
-                )
-            )
+        results = queryset.values(
+            *fields,
+            alternate_ids=Coalesce(F("subject__alternate_ids"), []),
+        ).annotate(
+            num_experiments=Count("biosamples__experiment"),
+            biosamples=Coalesce(ArrayAgg("biosamples__id", distinct=True, filter=Q(biosamples__id__isnull=False)), []),
+        )
+
+        # Get the biosamples with experiments data
+        phenopacket_ids = [result['subject_id'] for result in results]
+        biosamples_data = get_biosamples_with_experiment_details(phenopacket_ids)
+
+        # Group the biosamples data by subject_id
+        biosamples_with_experiments = defaultdict(list)
+        for b in biosamples_data:
+            biosamples_with_experiments[b["subject_id"]].append({
```

Comment:
i would prefer if this was again not called biosamples, since its actually more of 1 experiment per entry. since experiment records have biosample ID already, you could structure it more like

```
{
    ... all fields from experiment
    "biosample": {
         "id": "BIOSAMPLE_ID",
         "tissue": ...
     }
}
```

then this could be called `experiments_with_biosamples` since biosamples are attached for each experiment.